proxy/envoy/watcher.go

```diff
@@ -102,23 +102,28 @@ func (w *watcher) watchCerts(stop <-chan struct{}) {
 	if err != nil {
 		glog.Warning("failed to create a watcher for certificate files")
 	}
-	defer fw.Close()
+	defer func() {
```

Comment:
Flow cannot proceed to this point if NewWatcher returned non-nil error. The function should at least return though there is no easy way to propagate errors back in these watchers. Possibly later on we could plumb through context.Context for that purpose, but that is outside the scope of this PR.